packages/get-starknet/src/wallet.test.ts

```diff
@@ -128,18 +129,78 @@ describe('MetaMaskSnapWallet', () => {
   });
 
   describe('on', () => {
-    it('does nothing and not throw any error', async () => {
+    it('adds an event handler and starts polling if not already started', () => {
+      mockWalletInit({});
+
       const wallet = createWallet();
+      const handler = jest.fn();
 
-      expect(() => wallet.on('accountsChanged', jest.fn())).not.toThrow();
+      wallet.on('accountsChanged', handler);
+      expect((wallet as any).accountChangeHandlers.has(handler)).toBe(true);
+      wallet.off('accountsChanged', handler);
+    });
+
+    it('throws an error for unsupported events', () => {
+      const wallet = createWallet();
+      expect(() => wallet.on('unsupportedEvent' as any, jest.fn())).toThrow('Unsupported event: unsupportedEvent');
     });
   });
 
   describe('off', () => {
-    it('does nothing and not throw any error', async () => {
+    it('removes an event handler and stops polling if no handlers remain', () => {
+      mockWalletInit({});
+
       const wallet = createWallet();
+      const handler = jest.fn();
+
+      wallet.on('accountsChanged', handler);
+      wallet.off('accountsChanged', handler);
+
+      expect((wallet as any).accountChangeHandlers.has(handler)).toBe(false);
+    });
+
+    it('throws an error for unsupported events', () => {
+      const wallet = createWallet();
+      expect(() => wallet.off('unsupportedEvent' as any, jest.fn())).toThrow('Unsupported event: unsupportedEvent');
+    });
+  });
+
+  describe('event handling', () => {
+    it('triggers the accountsChanged handler when the event occurs', async () => {
+      const { getCurrentAccountSpy } = mockWalletInit({});
+      getCurrentAccountSpy.mockReset();
```

Comment:
```suggestion
      // Reset mocking of `getCurrentAccount` to allow for different results when triggered multiple times. 
      getCurrentAccountSpy.mockReset();
```

add this comment for better understand

or 

you put the new address as the default mocking address
and you use `mockResolvedValueOnce` to make sure the first call is the  initialAddress

that may be more clear